### PR TITLE
Switch to typer

### PR DESCRIPTION
**Pull request by znichollscr**
## Checklist

Please confirm that this pull request has done the following:

- [x] Tests added
- [ ] ~Documentation added (where applicable)~
- [x] Changelog item added to `changelog/`

### CODE REVIEW

**Comment by znichollscr** [comment]
@mikapfl can you take a quick look, just because you have used typer before

**Comment by znichollscr** [comment]
cc @lewisjared, looks pretty simple drop in for click

File: src/input4mips_validation/cli/__init__.py [hunk 1]
```diff
@@ -1,7 +1,48 @@
...
+    Entrypoint for the command-line interface
+    """
+    setup_logging()
+
+
+@app.command(name="validate-file")
+def validate_file_command(
+    filepath: Annotated[
+        Path, typer.Argument(exists=True, dir_okay=False, file_okay=True)
+    ],
```
Comment (mikapfl) [inline src/input4mips_validation/cli/__init__.py:27]: Cool, I didn't know about `Annotated`. Definitely clearer than the old way of writing
```
filepath: Path = typer.Argument(…)
```
which looked like `filepath` had a default value.

**Review by mikapfl — APPROVED** [review]

File: src/input4mips_validation/cli/__init__.py (hunk 1 above)
Comment (znicholls) [reply src/input4mips_validation/cli/__init__.py:27]: Nice, thanks!

Comment (znichollscr) [reply src/input4mips_validation/cli/__init__.py:27]: Nice, thanks!

**Pull request merged**